### PR TITLE
Remove unnecessary struct2interface dependency

### PR DESCRIPTION
**Pull request by agnivade**
#### Summary

Running go mod tidy on the repo removes it.
This was also preventing 1.14beta from running the repo,
because now it verifies modules.txt with go.mod, which had a mismatch.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -293,7 +293,18 @@ test-db-migration: start-docker ## Gets diff of upgrade vs new instance schemas.
...
+	@cp go.sum go.sum.orig
+	$(GO) mod tidy
+	@if [ "$$(diff go.mod go.mod.orig)" != "" -o "$$(diff go.sum go.sum.orig)" != "" ]; then \
+		echo "go.mod/go.sum was modified. diff- $(diff go.mod go.mod.orig) Run \"go mod tidy\"."; \
+		rm go.*.orig; \
+		exit 1; \
+	fi;
+	@rm go.*.orig;
+
+test-server: ## Runs tests.
```
Comment (hanzei) [inline Makefile]:
```suggestion
test-server: start-docker go-junit-report do-cover-file ## Runs tests.
```

File: Makefile [hunk 2]
```diff
@@ -293,7 +293,18 @@ test-db-migration: start-docker ## Gets diff of upgrade vs new instance schemas.
...
+gomodtidy:
+	@cp go.mod go.mod.orig
+	@cp go.sum go.sum.orig
+	$(GO) mod tidy
+	@if [ "$$(diff go.mod go.mod.orig)" != "" -o "$$(diff go.sum go.sum.orig)" != "" ]; then \
+		echo "go.mod/go.sum was modified. diff- $(diff go.mod go.mod.orig) Run \"go mod tidy\"."; \
+		rm go.*.orig; \
+		exit 1; \
+	fi;
+	@rm go.*.orig;
```
Comment (hanzei) [inline Makefile]: Maybe add `go.mod.orig` and `go.mod.orig` to `gitignore` just to make sure no one commits it?

File: Makefile [hunk 3]
```diff
@@ -293,7 +293,18 @@ test-db-migration: start-docker ## Gets diff of upgrade vs new instance schemas.
 	./scripts/mysql-migration-test.sh
 	./scripts/psql-migration-test.sh
 
-test-server: start-docker go-junit-report do-cover-file ## Runs tests.
+gomodtidy:
+	@cp go.mod go.mod.orig
+	@cp go.sum go.sum.orig
+	$(GO) mod tidy
+	@if [ "$$(diff go.mod go.mod.orig)" != "" -o "$$(diff go.sum go.sum.orig)" != "" ]; then \
+		echo "go.mod/go.sum was modified. diff- $(diff go.mod go.mod.orig) Run \"go mod tidy\"."; \
```
Comment (hanzei) [inline Makefile]: Does this output the diff correctly if only `go.sum` was modified?

**Comment by hanzei** [comment]
FYI: This fixes https://mattermost.atlassian.net/browse/IS-235

File: Makefile (hunk 3 above)
Comment (agnivade) [reply Makefile]: The diff was a check to see why the build was failing. I will update it properly.

**Comment by agnivade** [comment]
Cool, there was already a ticket for it.

**Comment by agnivade** [comment]
Couldn't figure out what is going on here. Filed: https://github.com/golang/go/issues/37376.

**Comment by agnivade** [comment]
@hanzei - I have removed the Makefile target for now until the Go team takes a look at it on Monday. Let's push this in master for now, to let Go 1.14 run the codebase ?

File: go.mod [hunk 4]
```diff
@@ -66,8 +66,6 @@ require (
 	github.com/pelletier/go-toml v1.6.0 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.4.0
-	github.com/prometheus/client_model v0.2.0
```
Comment (hanzei) [inline go.mod]: Hm, when I run `go mod tidy` locally, this is includes. Any ideas why?

```
$ go version
go version go1.13.7 linux/amd64
```

File: go.mod [hunk 5]
```diff
@@ -41,7 +41,7 @@ require (
 	github.com/icrowley/fake v0.0.0-20180203215853-4178557ae428
 	github.com/jaytaylor/html2text v0.0.0-20190408195923-01ec452cbe43
 	github.com/jmoiron/sqlx v1.2.0
-	github.com/jonboulle/clockwork v0.1.0
```
Comment (hanzei) [inline go.mod]: And also this line.

File: Makefile [hunk 6]
```diff
@@ -293,6 +293,17 @@ test-db-migration: start-docker ## Gets diff of upgrade vs new instance schemas.
 	./scripts/mysql-migration-test.sh
 	./scripts/psql-migration-test.sh
 
+gomodtidy:
+	@cp go.mod go.mod.orig
+	@cp go.sum go.sum.orig
+	$(GO) mod tidy
+	@if [ "$$(diff go.mod go.mod.orig)" != "" -o "$$(diff go.sum go.sum.orig)" != "" ]; then \
+		echo "go.mod/go.sum was modified. \ndiff- $$(diff go.mod go.mod.orig) \n$$(diff go.sum go.sum.orig) \nRun \"go mod tidy\"."; \
```
Comment (hanzei) [inline Makefile]: Apparently `\n` doesn't work here:
```bash
go.mod/go.sum was modified. \ndiff- 44c44
< 	github.com/jonboulle/clockwork v0.1.0
---
> 	github.com/jonboulle/clockwork v0.1.0 // indirect
69d68
< 	github.com/prometheus/client_model v0.2.0 \n5,6d4
< contrib.go.opencensus.io/exporter/ocagent v0.4.9/go.mod h1:ueLzZcP7LPhPulEBukGn4aLh7Mx9YJwpVJ9nL2FYltw=
< git.apache.org/thrift.git v0.0.0-20180902110319-2566ecd5d999/go.mod h1:fPE2ZNJGynbRyZ4dJvy6G277gSllfV2HJqblrnkyeyg= \nRun "go mod tidy".
```

File: go.mod (hunk 4 above)
Comment (agnivade) [reply go.mod]: That's what the bug is about which I filed. If you run go mod tidy, it changes go.mod. But if you run go mod vendor again, it reverts those changes. They are in conflict with each other. I just went with one version.

File: Makefile (hunk 6 above)
Comment (agnivade) [reply Makefile]: Strange, this is what I see:
```
go mod tidy
warning: ignoring symlink /home/agniva/mattermost/mattermost-server/client
warning: ignoring symlink /home/agniva/mattermost/mattermost-server/enterprise
go.mod/go.sum was modified. 
diff- 44c44
< 	github.com/jonboulle/clockwork v0.1.0
---
> 	github.com/jonboulle/clockwork v0.1.0 // indirect
69d68
< 	github.com/prometheus/client_model v0.2.0 
 
Run "go mod tidy".
make: *** [Makefile:300: gomodtidy] Error 1
```

**Comment by agnivade** [comment]
@hanzei - This is now blocking Go 1.14 to run the codebase. Should we push this ?

File: Makefile (hunk 6 above)
Comment (hanzei) [reply Makefile]: What shell are you running? I'm using `bash`

Comment (hanzei) [reply Makefile]: What shell are you running? I'm using `bash`

File: go.mod (hunk 4 above)
Comment (hanzei) [reply go.mod]: Got it. Thanks for point out that this is caused by https://github.com/golang/go/issues/37376

File: Makefile (hunk 6 above)
Comment (agnivade) [reply Makefile]: I'm also running bash.

**Comment by agnivade** [comment]
> I'm fine with merging this now as we come back to and enforce gomodtidy

Sure, I am looking into it.

**Pull request merged**